### PR TITLE
Feature/rr 385 add processed to application

### PR DESCRIPTION
**Pull request by CaraHill**
https://govtnz.atlassian.net/browse/RR-385

This is the first part of the work for this story - refactoring the backend code so RebateForms use `status` instead of `completed` and are aware of the two statuses, `signed` and `not signed`.

All is working and tests are passing, so it is in a good state to complete the work on the rebate form state machine and add the processed status.

I've also deleted a bunch of unused files and code - a bit of extra clean up.

# How to test this change

- [ ] has tests
- [ ] at least one a reviewer ran the code

### CODE REVIEW

File: app/models/rebate_form.rb [hunk 1]
```diff
@@ -16,13 +16,16 @@ class RebateForm < ApplicationRecord
   validates :property, presence: true
 
   validate :same_council
-  validate :only_completed_forms_in_batch
+  # validate :only_completed_forms_in_batch
```
Comment (Br3nda) [inline app/models/rebate_form.rb]: do we need this for batches?

File: db/migrate/20190527231403_add_status_and_remove_completed_from_rebate_form.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+class AddStatusAndRemoveCompletedFromRebateForm < ActiveRecord::Migration[5.2]
+  def up
+    add_column :rebate_forms, :status, :string, default: 'not signed'
+    update_existing_rebate_forms_up
+    remove_column :rebate_forms, :completed
```
Comment (Br3nda) [inline db/migrate/20190527231403_add_status_and_remove_completed_from_rebate_form.rb]: Are there any existing applications we need to migrate here? I had a mixture of completed and not completed applications, but now they're all not signed.
If we run this migration on UAT all applications will be considered not signed, even though they do have signatures.

Comment (CaraHill) [reply db/migrate/20190527231403_add_status_and_remove_completed_from_rebate_form.rb]: @Br3nda did you see the code for `update_existing_forms_up'? It should update all existing forms, and on my local it is respects the mix of completed and not.

Comment (mischa-s) [reply db/migrate/20190527231403_add_status_and_remove_completed_from_rebate_form.rb]: Worth considering - on both my and Cara's machines this migration does preserve that status as intended. Current understanding is there is no data on UAT which must be preserved but lets check with @robholmesnz before we deploy

**Comment by Br3nda** [comment]
would look better without wrapping here.
image

File: app/javascript/components/SummaryTabs.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (Br3nda) [reply app/javascript/components/SummaryTabs.js]: I can have a look at why it's doing this. Sorry it's being noisy

**Comment by mischa-s** [comment]
it doesn't wrap on my machine, percy, or cara's machine - what config are you using to wrap it?

File: db/migrate/20190527231403_add_status_and_remove_completed_from_rebate_form.rb (hunk 2 above)
Comment (Br3nda) [reply db/migrate/20190527231403_add_status_and_remove_completed_from_rebate_form.rb]: I'm rolling back and retesting the migration

Comment (Br3nda) [reply db/migrate/20190527231403_add_status_and_remove_completed_from_rebate_form.rb]: My mistake -- they all migrated BEAUTIFULLY  :rose:

**Comment by mischa-s** [comment]
image

File: app/javascript/containers/CustomersSummary.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    }
+  }
+
+  onSubmit(values = {}) {
+    this.fetchRebates('not signed', values.name);
+  }
+
+  render() {
+    const { applicationState, rebateForms } = this.state;
+    console.log(this.state, 'this state');
```
[the remark this answers is not in the document]
Comment (Br3nda) [reply app/javascript/containers/CustomersSummary.js]: Do we want to keep the console here?

**Comment by Br3nda** [comment]
I'm using firefox. I think percy uses chrome headless.

File: db/migrate/20190527231403_add_status_and_remove_completed_from_rebate_form.rb (hunk 2 above)
Comment (CaraHill) [reply db/migrate/20190527231403_add_status_and_remove_completed_from_rebate_form.rb]: @Br3nda Oh good, that's a relief! 🎉

**Comment by Br3nda** [comment]
Don't worry about chasing down the wrappy - i see it's fine enough in chrome.

**Comment by mischa-s** [comment]
image

percy also uses firefox ^

**Comment by CaraHill** [comment]
Thanks @Br3nda! 😄

**Pull request merged**